src/index.js

```diff
@@ -23,4 +46,136 @@ export default class OsmRequest {
+      }
+    };
+  }
+
+  /**
+   * Fetch an OSM element by its ID
+   * @param {string} osmId Eg: node/12345
+   * @return {Promise}
+   */
+  fetchElement(osmId) {
+    return fetchElementRequest(this.endpoint, osmId);
+  }
+
+  /**
+   * Add or replace a property in a given element
+   * @param {object} element A geoJSON element
+   * @param {string} propertyName
+   * @param {string} propertyValue
+   * @return {object} A new version of the geoJSON element
+   */
+  setProperty(element, propertyName, propertyValue) {
+    throwIfNotPoint(element);
+
+    const newElement = realObjectCopy(element);
+    newElement.properties[propertyName] = propertyValue;
+    return newElement;
+  }
+
+  /**
+   * Add or replace several properties in a given element
+   * @param {object} element A geoJSON element
+   * @param {object} properties
+   * @return {object} A new version of the geoJSON element
+   */
+  setProperties(element, properties) {
+    throwIfNotPoint(element);
+
+    const newElement = realObjectCopy(element);
+    const clonedProperties = realObjectCopy(properties);
+
+    for (const propertyName of Object.keys(clonedProperties)) {
+      newElement.properties[propertyName] = clonedProperties[propertyName];
+    }
+
+    return newElement;
+  }
+
+  /**
+   * Remove a property from a given element
+   * @param {object} element A geoJSON element
+   * @param {string} propertyName
+   * @return {object} A new version of the geoJSON element
+   */
+  removeProperty(element, propertyName) {
+    throwIfNotPoint(element);
+
+    const newElement = realObjectCopy(element);
+    delete newElement.properties[propertyName];
+    return newElement;
+  }
+
+  /**
+   * Replace the coordinates of the OSM node an return a copy of the element
+   * @param {object} element
+   * @param {number} lat
+   * @param {number} lon
+   * @return {object} A new version of the geoJSON element
+   */
+  setCoordinates(element, lat, lon) {
+    throwIfNotPoint(element);
+
+    const newElement = realObjectCopy(element);
+    newElement.geometry.coordinates = [lon, lat];
+    return newElement;
+  }
+
+  /**
+   * Set the current UTC date to a given element
+   * @param {object} element
+   * @return {object} A new version of the geoJSON element
+   */
+  setTimestampToNow(element) {
+    return this.setProperty(element, 'timestamp', new Date().toISOString());
+  }
+
+  /**
+   * Increase the version number of an element
+   * @param {object} element
+   * @return {object} A new version of the geoJSON element
+   */
+  incrementVersion(element) {
+    const currentVersion = parseInt(element.properties.version || 0, 10);
+    return this.setProperty(
+      element,
+      'version',
+      (currentVersion + 1).toString()
+    );
+  }
+
+  /**
+   * Send a request to OSM to create a new changeset
+   * @param {string} [author]
+   * @param {string} [comment]
+   * @return {Promise}
+   */
+  createChangeset(author = '', comment = '') {
```

Comment:
`author` does not seem an appropriate variable name: this is supposed to represent the computer program (editor or script) which made the changes, not an author.

I think this value should be set as a config at instantiation time, as we don't want to change it between edits.

And we should provide a proper default value ("osm-request javascript lib" or something like that).

Reply:
`author` is now replaced by `createdBy`, as it is called in the final xml (`created_by`).

@PanierAvide and I added `created_by:library` too, to tell that the changeset has been created by OSM Request x.x.x.

Reply:
The `created_by:library`is a good idea :+1: 

I still think we should let the user set the `created_by` at the creation of the OsmRequest object, and not when a new changeset is opened. This value should not change between edits, and should identify his own app (OpenBeerMap / MapContrib / Jungle Bot / etc)